tests/Feature/auth/RegisterControllerTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Tests\Feature\Auth;
+
+use App\Models\User;
+use Carbon\Carbon;
+use Illuminate\Auth\Notifications\VerifyEmail;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class RegisterControllerTest extends TestCase
+{
+    use RefreshDatabase;
+    /** @test */
+    public function user_can_register_into_app_and_is_logged_in_after(): void
+    {
+        $email = 'mail@example.com';
+        $pass = 'example_pass123';
+        $credentials = [
+            'email' => $email,
+            'password' => $pass,
+        ];
+
+        $newUser = User::factory()->make($credentials);
+        $newUser = $newUser->toArray();
+        $newUser['password'] = $pass;
+        $newUser['password_confirmation'] = $pass;
+        $response = $this->postJson(route('register'), $newUser);
+        $response->assertStatus(201)
+            ->assertJsonFragment(['name' => $newUser['name']])
+            ->assertJsonFragment(['email' => $newUser['email']]);
+
+    }
+
+    /** @test */
+    public function user_can_verify_his_email_after_registration(): void
+    {
+        $email = 'mails@example.com';
+        $pass = 'example_pass123';
+        $name = 'Joe Doe';
+        $credentials = [
+            'name' => $name,
+            'email' => $email,
+            'password' => $pass,
+            'password_confirmation' => $pass,
+        ];
+
+        $response = $this->postJson(route('register'), $credentials);
+        $response->assertStatus(201);
+
+        $response = $this->postJson(route('login'), $credentials);
+        $response->assertRedirect('home');
```

Comment:
Żaden endpoint API nie powinien powodować przekierowania, API raczej z reguły nie robią przekierowań bo trudno to obsłużyć.